text/0000-rust-governance.md

```diff
@@ -0,0 +1,728 @@
+  (Thanks to Nick Cameron for this suggestion.)
+
+Subteams make it possible to involve a larger, more diverse group in the
+decision-making process. In particular, **they should involve a mix of**:
+
+* Rust project leadership, in the form of at least one core team member (the
+  leader of the subteam).
+
+* Area experts: people who have a lot of interest and expertise in the subteam
+  area, but who may be far less engaged with other areas of the project.
+
+* Stakeholders: people who are strongly affected by decisions in the
+  subteam area, but who may not be experts in the design or
+  implementation of that area. *It is crucial that some people heavily
+  using Rust for applications/libraries have a seat at the table, to
+  make sure we are actually addressing real-world needs.*
+
+Members should have demonstrated a good sense for design and dealing with
+tradeoffs, an ability to work within a framework of consensus, and of course
+sufficient knowledge about or experience with the subteam area. Leaders should
+in addition have demonstrated exceptional communication, design, and people
+skills. They must be able to work with a diverse group of people and help lead
+it toward consensus and execution.
+
+Each subteam is led by a member of the core team. **The leader is responsible for**:
+
+* Setting up the subteam:
+
+    * Deciding on the initial membership of the subteam (in consultation with
+      the core team). Once the subteam is up and running.
+
+    * Working with subteam members to determine and publish subteam policies and
+      mechanics, including the way that subteam members join or leave the team
+      (which should be based on subteam consensus).
+
+* Communicating core team vision downward to the subteam.
+
+* Alerting the core team to subteam RFCs that need global, cross-cutting
+  attention, and to RFCs that have entered the "final comment period" (see below).
+
+* Ensuring that RFCs and PRs are progressing at a reasonable rate, re-assigning
+  shepherds/reviewers as needed.
+
+* Making final decisions in cases of contentious RFCs that are unable to reach
+  consensus otherwise (should be rare).
+
+The way that subteams communicate internally and externally is left to each
+subteam to decide, but:
+
+* Technical discussion should take place as much as possible on public forums,
+  ideally on RFC/PR threads and tagged discuss posts.
+
+* Each subteam will have a dedicated
+  [discuss forum](http://internals.rust-lang.org/) tag.
+
+* Subteams should actively seek out discussion and input from stakeholders who
+  are not members of the team.
+
+* Subteams should have some kind of regular meeting or other way of making
+  decisions. The content of this meeting should be summarized with the rationale
+  for each decision -- and, as explained below, decisions should generally be
+  about weighting a set of already-known tradeoffs, not discussing or
+  discovering new rationale.
```

Comment:
Note the bullet in the RFC:

> Technical discussion should take place as much as possible on public forums, ideally on RFC/PR threads and tagged discuss posts.

I'll supplement this by introducing public, logged IRC channels as well. There are several existing channels that can be repurposed for this role.

Also, I personally don't think that literal *transcripts* of meetings are always so valuable -- they are often inaccurate, and take one person out of the conversation, and can be very hard to decipher. I think it would be much better to instead coherently summarize what was said and done in any meetings -- the traditional notion of "minutes".

In my experience meetings are actually a poor place to make decisions -- it's easy to be swayed by the "heat of the moment" and a sense that an opinion needs to be registered immediately. It works much better to log objections directly onto RFC threads, and then follow up as those objections are addressed.

In the core team, for example, we almost never make decisions on RFCs in a meeting; at most, we call attention to RFCs that need decisions soon. Instead, we keep track of RFCs nearing decision point (a bit like the "final comment period"), and make sure that people who have objections have raised them publicly on thread. I would suggest a similar process for subteams.

As an aside, I think private conversation has its place. It allows for greater bandwidth (due to shared context), greater candor, and makes it easier to hash out ideas in an earlier phase, and so on. I do not think it's possible or desirable to legislate away its existence.

Rather, there needs to be a culture of public communication:

* As soon as an opinion or idea becomes relatively formed -- ready for public consumption -- go public with it. Whenever practical, start in public.

* All decisions are made based on previously-discussed public rationale.

* Use of processes like RFCs to move things forward in a very public, visible way.

All of this is already in the RFC.